android/libraries/rib-badoo/src/main/java/com/badoo/common/rib/BaseInteractor.kt

```diff
@@ -0,0 +1,89 @@
+package com.badoo.common.rib
+
+import android.arch.lifecycle.Lifecycle
+import android.arch.lifecycle.LifecycleOwner
+import android.arch.lifecycle.LifecycleRegistry
+import android.os.Bundle
+import android.support.annotation.CallSuper
+import com.badoo.mvicore.android.lifecycle.CreateDestroyBinderLifecycle
+import com.badoo.mvicore.android.lifecycle.ResumePauseBinderLifecycle
+import com.badoo.mvicore.android.lifecycle.StartStopBinderLifecycle
+import com.badoo.mvicore.binder.Binder
+import com.uber.rib.core.Interactor
+import com.uber.rib.core.RibAndroidView
+import io.reactivex.disposables.Disposable
+
+abstract class BaseInteractor<V : RibAndroidView, R : BaseViewRouter<V>>(
+    private val disposables: List<Disposable>
+) : Interactor<V>(), LifecycleOwner {
+
+    private val ribLifecycleRegistry = LifecycleRegistry(this)
+    private val viewLifecycleRegistry = LifecycleRegistry(this)
+
+    final override fun didBecomeActive(savedInstanceState: Bundle?) {
+        super.didBecomeActive(savedInstanceState)
+        ribLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_CREATE)
+        didBecomeActive(ribLifecycleRegistry, savedInstanceState)
+    }
+
+    open fun didBecomeActive(ribLifecycle: Lifecycle, savedInstanceState: Bundle?) {
+    }
+
+    @CallSuper
+    fun onViewCreated() {
+        viewLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_CREATE)
+        // fixme
+        onViewCreated((router as BaseViewRouter<V>).view!!, viewLifecycleRegistry)
+    }
+
+    open fun onViewCreated(view: V, viewLifecycle: Lifecycle) {
+    }
+
+    fun onViewDestroyed() {
+        viewLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+    }
+
+    fun onStart() {
+        ribLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_START)
+        viewLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_START)
+    }
+
+    // todo call this when removed from view
+    fun onStop() {
+        ribLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_STOP)
+        viewLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_STOP)
+    }
+
+    fun onResume() {
+        ribLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_RESUME)
+        viewLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_RESUME)
+    }
+
+    // todo call this when removed from view
+    fun onPause() {
+        ribLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_PAUSE)
+        viewLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_PAUSE)
+    }
+
+    override fun willResignActive() {
+        super.willResignActive()
+        viewLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY) // todo probably this is not needed?
+        ribLifecycleRegistry.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY)
```

Comment:
I guess it means create rib / create view respectively, and I see no other way to do it, since Lifecycle.Event doesn't contain a different element for them. Any better idea?